### PR TITLE
add validations for  metric and label names

### PR DESCRIPTION
**Pull request by imakhlin**
1. validate metric name
2. validate label name

### CODE REVIEW

File: pkg/utils/validators.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+const (
+	metricNameValidationRegexStr = `^[a-zA-Z_:]([a-zA-Z0-9_:])*$`
+	labelValidationRegexStr      = `^[a-zA-Z_]([a-zA-Z0-9_])*$`
```
Comment (gtopper) [inline pkg/utils/validators.go]: Any particular reason we support colon in name but not label?

File: pkg/tsdbctl/add.go [hunk 2]
```diff
@@ -248,6 +252,10 @@ func strToLabels(name, lbls string) (utils.Labels, error) {
 			if len(splitLbl) != 2 {
 				return nil, errors.New("labels must be in the form: key1=label1,key2=label2,...")
 			}
+
+			if err := utils.IsValidLabelName(splitLbl[0]); err != nil {
+				return nil, errors.Wrap(err, fmt.Sprintf("illegal lable name: '%s'", splitLbl[0]))
```
Comment (gtopper) [inline pkg/tsdbctl/add.go]: Typo. Do we really need to wrap it anyway? There's no information here that's not already in `err`.

Comment (imakhlin) [reply pkg/tsdbctl/add.go]: Wrapping it to create better stack trace.

File: pkg/utils/validators.go (hunk 1 above)
Comment (imakhlin) [reply pkg/utils/validators.go]: No, this is what Prometheus do. We just mimic same convention.
see: https://github.com/prometheus/client_java/issues/28

**Pull request merged**